torch/_dynamo/variables/functions.py

```diff
@@ -540,13 +540,16 @@ def has_force_unpack_var_sequence(self, tx) -> builtins.bool:
 
     def force_unpack_var_sequence(self, tx) -> list[VariableTracker]:
         result = []
+        self.force_apply_to_var_sequence(tx, result.append)
+        return result
+
+    def force_apply_to_var_sequence(self, tx, fn) -> None:
         while True:
             try:
-                result.append(self.next_variable(tx))
+                fn(self.next_variable(tx))
```

Comment:
If I understood correctly, you apply `fn` in each element of the generator until it gets exhausted. Wouldn't `map(fn, self.force_unpack_var_sequence(tx))`, lead to the same behavior?

Reply:
We need to apply fn as we extract elements from the generator/iterator. The suggestion would result in the old behavior because force_unpack_var_sequence causes immediate unpacking - it does not return a lazy iterator.